app/src/app/components/providers/ThemeProvider.tsx

```diff
@@ -15,31 +25,70 @@ export const useTheme = () => {
   return context;
 };
 
-export const ThemeProvider = ({ children, defaultTheme = "light", ...props }: ThemeProviderProps) => {
+const getSystemTheme = (): Theme => {
+  if (window.matchMedia("(prefers-color-scheme: dark)")?.matches) {
+    return "dark";
+  } else if (window.matchMedia("(prefers-color-scheme: light)")?.matches) {
+    return "light";
+  } else {
+    return privilegedTheme;
+  }
+};
+
+export const ThemeProvider = ({ children, ...props }: ThemeProviderProps) => {
+  const { brandingConfig, isLoading, error } = useGetBrandingConfig();
+
+  if (error) return <ErrorComponent message={error.message} error={error} />;
+
   const [theme, setTheme] = useState<Theme>(
     () => (localStorage.getItem(LocalStorageKeys.THEME) as Theme) || defaultTheme
   );
 
+  const [availableThemes, setAvailableThemes] = useState<Theme[]>([]);
+
+  useEffect(() => {
+    if (!isLoading && brandingConfig) {
+      if (brandingConfig.darkModeEnabled && !brandingConfig.lightModeEnabled) {
+        setAvailableThemes(["dark"]);
+      } else if (brandingConfig.lightModeEnabled && !brandingConfig.darkModeEnabled) {
+        setAvailableThemes(["light"]);
+      } else {
+        setAvailableThemes(defaultAvailableThemes);
+      }
+    }
+  }, [brandingConfig, isLoading]);
+
+  // Work out if the theme we're thinking of applying is available, and update the theme if not.
+  const verifyThemeIsAvailable = () => {
+    const proposedTheme = theme === "system" ? getSystemTheme() : theme;
+
+    if (!availableThemes.includes(proposedTheme) && availableThemes.length !== 0) {
+      setTheme(availableThemes[0]);
+    }
+  };
+
+  useEffect(verifyThemeIsAvailable, [availableThemes]);
+
   useEffect(() => {
     const root = window.document.documentElement;
 
-    root.classList.remove("light", "dark");
+    root.classList.remove(...defaultAvailableThemes);
 
     if (theme === "system") {
-      const systemTheme = window.matchMedia("(prefers-color-scheme: dark)")?.matches ? "dark" : "light";
-
+      const systemTheme = getSystemTheme();
       root.classList.add(systemTheme);
-      return;
+    } else {
+      root.classList.add(theme);
     }
-
-    root.classList.add(theme);
   }, [theme]);
 
   const value = {
+    availableThemes,
     theme,
-    setTheme: (theme: Theme) => {
-      localStorage.setItem(LocalStorageKeys.THEME, theme);
+    setCurrentTheme: (theme: Theme) => {
+      verifyThemeIsAvailable();
```

Comment:
`setTheme` is being called 2 times if it is called in `verifyThemeIsAvailable`.... call once only 1